examples/plot_multiple_imputation.py

```diff
@@ -0,0 +1,472 @@
+"""
+=================================================
+Imputing missing values using multiple imputation
+=================================================
+
+By default, the IterativeImputer performs single imputation: a method where
+every missing value is replaced with one imputed value. The chained character
+of the method and the possiblity to draw imputation values from the posterior
+distribution of a Bayesian imputation model allows for the finding of unbiased
+statistical estimates. However, the disadvantage is that every imputed value is
+treated as if the value was observed, leading to an imputed dataset that does
+not reflect the uncertainty that occurs due to the presence of missing values.
+This makes it hard to find valid statistical inferences because the variance
+(and standard error) of statistical estimates become too small.
+
+An alternative is using the IterativeImputer to perform multiple imputation: a
+method where every missing value is imputed multiple times. The procedure
+results in multiple datasets where the observed data is similar in every
+dataset, but the imputed data is different. All desired steps after imputation
+are performed on every dataset, including the analysis. Then, Rubin's pooling
+rules are used to combine the estimates into one final result.
+
+In this example we will show how to use the ITerativeImputer to perform
+multiple imputation, what the effect is on the standard error of beta
+coefficients and how to set up a prediction model using multiple imputation.
+"""
+
+import math
+import numpy as np
+import matplotlib.pyplot as plt
+from scipy import stats
+
+from sklearn.datasets import load_boston
+from sklearn.linear_model import LinearRegression
+from sklearn.preprocessing import StandardScaler
+from sklearn.model_selection import train_test_split
+from sklearn.impute import ChainedImputer
+from sklearn.metrics import mean_squared_error as mse
+
+rng = np.random.RandomState(0)
+
+
+# Start by defining a basic amputation function
+def ampute(X, missing_rate=0.75, mech="MCAR"):
+    n_samples = X.shape[0]
+    n_features = X.shape[1]
+    X_incomplete = X.copy()
+
+    # MCAR mechanism
+    if mech == 'MCAR':
+        for i in np.arange(n_features):
+            dropped_indices = np.array(np.random.choice(np.arange(n_samples),
+                                                        size=int(missing_rate
+                                                                 * n_samples),
+                                                        replace=False))
+            X_incomplete[dropped_indices[:, None], i] = None
+
+    # MNAR mechanism
+    if mech == "MNAR":
+        for i in np.arange(n_features):
+            data_values = -np.mean(X[:, i]) + X[:, i]
+            weights = list(map(lambda x: math.exp(x) / (1 + math.exp(x)),
```

Comment:
why is this not just `np.exp(data_values) / (1 + np.exp(data_values))` (or `1 / (1 + np.exp(-data_values))`)?

Reply:
Think you're absolutely right. Changed it.